File: Entrega 1/NumericMethodsJava/MultipleRoots.java
package NumericMethods;

/**
 *
 * @author Valeria
 */
public class MultipleRoots {

    public static double f(double x) {
        //h(x) =e^x - x - 1
        return (Math.pow((Math.exp(1)), x)) - x - 1;
    }

    public static double df1(double x) {
        //h'(x) =e^x - 1
        return (Math.pow((Math.exp(1)), x)) - 1;
    }

    public static double df2(double x) {
        //h''(x) =e^x
        return (Math.pow((Math.exp(1)), x));
    }

    public static void roots(double x0, double tolerance, int iterations) {

        double resultf = f(x0);
        double resultf1 = df1(x0);
        double resultf2 = df2(x0);
        int i = 0;

        //Method data input control
        if (iterations < 1) {
            System.out.println("Iterations must be greater than 0");
        }
        if (resultf == 0) {
            System.out.println(x0 + " Its a root");
        }

        //First iteration (iteration 0)
        System.out.println("Iter: " + i);
        System.out.println("x0: " + x0);
        System.out.println("f(xi): " + f(x0));
        System.out.println("E: ");
        System.out.println("");
        i++;

        double xi = x0 - ((resultf * resultf1) / (Math.pow(resultf1, 2) - (resultf * resultf2)));
        System.out.println("Luego de hacer la super operacion: " + Math.abs(xi - x0));
        //Loop start
        while (resultf != 0 && i <= iterations && Math.abs(xi - x0) > tolerance) {
            double error = Math.abs(xi - x0);
            x0 = xi;
            resultf = f(x0);
            resultf1 = df1(x0);
            resultf2 = df2(x0);
            xi = x0 - ((resultf * resultf1) / (Math.pow(resultf1, 2) - (resultf * resultf2)));

            System.out.println("Iter: " + i);
            System.out.println("xi: " + x0);
            System.out.println("f(xi): " + resultf);
            System.out.println("E: " + error);
            System.out.println("");
            i++;
        }

        //Method data output control
        if (resultf == 0) {
            System.out.println("An approximation of the root was found in " + (xi));
        } else if (i > iterations) {
            System.out.println("Iteration limit reached");
        } else {
            if (Math.abs(xi - x0) < tolerance) {
                System.out.println("The maximum tolerance permitted " + (tolerance) + ". In the iteration " + (i) + " the maximum tolerance was reached " + (Math.abs(xi - x0)) + "\n");
                System.out.println("Data in the last iteration");
                System.out.println("Iteration " + (i - 1));
                System.out.println("x0= " + (x0));
                System.out.println("f(x0)= " + f(x0));
                System.out.println("E= " + (Math.abs(xi - x0)));
            }
        }
    }
}

File: Entrega 1/NumericMethodsPython/MultipleRoots.py

import math

# author Valeria

def f(x):
    #return h(x) =e^x - x - 1
    return (math.pow((math.exp(1)), x)) - x - 1

def df1(x):
    #return h'(x) =e^x - 1
    return (math.pow((math.exp(1)), x)) - 1

def df2(x):
    #return h''(x) =e^x
    return (math.pow((math.exp(1)), x))

def roots(x0, tolerance, iterations):

    resultf = f(x0)
    resultf1 = df1(x0)
    resultf2 = df2(x0)
    i = 0;

    #Method data input control
    if iterations < 1:
        print("Iterations must be greater than 0")

    if resultf == 0:
        print(x0 + "Its a root")
    
    #First iteration (iteration 0)
    print("Iter: " , i)
    print("x0: " ,x0)
    print("f(xi): " , f(x0))
    print("E: ")
    print("")
    i += 1
    xi = x0 - ((resultf * resultf1) /(math.pow(resultf1, 2) - (resultf * resultf2)))

    #Loop start
    while(resultf != 0 and i <= iterations and abs(xi - x0) > tolerance):
        error = abs(xi - x0)
        x0 = xi
        resultf = f(x0)
        resultf1 = df1(x0)
        resultf2 = df2(x0)
        xi = x0 - ((resultf * resultf1) / (math.pow(resultf1, 2) - (resultf * resultf2)))

        print("Iter: " , i)
        print("x0: " , x0)
        print("f(xi): " , resultf)
        print("E: " , error)
        print("")
        i += 1

    if resultf == 0:
        print("An approximation of the root was found in " , (xi))
    elif i > iterations:
        print("Iteration limit reached")
    elif (abs(xi - x0) < tolerance):
        print("The maximum tolerance permitted " , (tolerance) , ". In the iteration " , (i) , " the maximum tolerance was reached " , abs(xi - x0))
        print("Data in the last iteration")
        print("Iteration " , (i - 1))
        print("x0= " , (x0))
        print("f(x0)= " + f(x0))
        print("E= " , (abs(xi - x0)))


roots(1, math.pow(10, -7), 100)